receiver/awslambdareceiver/handler.go

```diff
@@ -77,12 +135,23 @@ func (s *s3Handler[T]) handle(ctx context.Context, event json.RawMessage) error
 		return err
 	}
 
-	data, err := s.s3Unmarshaler(body)
-	if err != nil {
-		return fmt.Errorf("failed to unmarshal logs: %w", err)
+	var data T
+	if s.s3Unmarshaler == nil {
+		switch any(s.emitType).(type) {
```

Comment:
This is a bit hacky, type-asserting on a zero value when we should already know from s3Handler's type parameter. Rather than doing this, can we just arrange for `s3Unmarshaler` to always be non-nil?

When creating the s3Handler for logs and an encoding hasn't been specified, we can create an unmarshaler that uses bytesToPlogs.

Reply:
Depending on the outcome of https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/44562#pullrequestreview-3570161686, maybe we could remove all default unmarshalling code from the receiver, and:
 - default to using the "text_encoding" extension for S3 logs
 - default to using the "awscloudwatchmetrics_encoding" extension for S3 metrics
 - default to using the "awslogs_encoding" extension with `format: cloudwatch` for CloudWatch Logs